File: 3096/3096.pas
{$mode objfpc}{$H+}

uses
  math;

var
  n: integer;
  r: integer;
begin
  readln(n);
  r := floor(n*log10(n/exp(1)) + log10(2*pi*n)/2 + 1);
  writeln(r);
end.

File: 3096/3096.kt
import java.util.Scanner;
import kotlin.math.floor;
import kotlin.math.log10;

fun main(args: Array<String>) {
    val scanner = Scanner(System.`in`);
    
    val n: Int = scanner.nextInt();
    
    val r: Double = floor(n*log10(n/Math.E) + log10(2*Math.PI*n)/2.0) + 1;
    println(r.toInt());
}